raft-tech/TANF-app
# Issue 966: Add a Circle CI testing step that confirms no secret keys have been committed

### Summary of Changes

This PR adds an additional step to the Circle CI `build_and_test` workflow to perform a scan with [TruffleHog](https://github.com/trufflesecurity/truffleHog) and ensure that no secret keys have been committed anywhere in the git repository.

Closes #966
_Acceptance criteria as stated in the issue_

### How to Test

- Ensure the `secrets-check` step in the `build_and_test` job in CircleCI runs the `trufflehog-check` script.
- Ensure the script exits with no issues.
- Checkout this branch and create a file with fake credentials like `JWT_KEY=THISISAREALTOKEN`, commit, and push that file.
- Ensure the `secrets-check` step fails.
- Undo the commit and clean up the PR branch again.

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

	- [x] Circle CI builds have a step that confirms no secret keys have been committed to the repo

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

This PR does not change application code, and test coverage does not apply.

> This PR does not introduce frontend or backend application code.

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*


This PR does not introduce frontend or backend application code.

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

This PR does not include a11y relevent application code changes.

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging


### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

No application code added or changed. No dependencies added.

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

	- None detected.
